app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabase.kt

```diff
@@ -382,6 +463,200 @@ class FirebaseDatabase : Database() {
         return Utils.failedFuture(Error("Not implemented"))
     }
 
+    override fun createChatConversation(
+        name: String,
+        membersList: List<String>,
+        creatorId: String,
+        welcomeMessage: String,
+    ): CompletableFuture<Unit> {
+        // create the id of the new conversation
+        val pushedPostRef: DatabaseReference = chatsRoot().push()
+        val conversationId: String? = pushedPostRef.key
+
+        if (conversationId == null) {
+            throw Exception("Error in the generation of the conversation id !")
```

Comment:
Suggestion: Perhaps you can return a future that completes Exceptionally instead of an exception
val future = CompletableFuture<Unit>()
            future.completeExceptionally(Exception("Error in the generation of the conversation id!"))
            return future